dali/image/bmp.cc

```diff
@@ -78,14 +78,15 @@ BmpImage::BmpImage(const uint8_t *encoded_buffer, size_t length, DALIImageType i
 
 Image::Shape BmpImage::PeekShapeImpl(const uint8_t *bmp, size_t length) const {
   DALI_ENFORCE(bmp != nullptr);
+  DALI_ENFORCE(length >= 18);
   auto ptr = bmp + 14;
   uint32_t header_size = ConsumeValue<uint32_t>(ptr);
   int64_t h = 0, w = 0, c = 0;
   int bpp = 0, compression_type = BMP_COMPRESSION_RGB;
   const uint8_t* palette_start = nullptr;
   size_t ncolors = 0;
   size_t palette_entry_size = 0;
-  if (length >= 22 && header_size == 12) {
+  if (length >= 24 && header_size == 12) {
```

Comment:
```suggestion
  if (length >= 26 && header_size == 12) {
```
?
We advance by 8 bytes, starting at 18.

Reply:
18+2( w = ConsumeValue<uint16_t>(ptr)) + 2( ConsumeValue<uint16_t>(ptr);) + 2(ptr += 2;) +2(bpp = ConsumeValue<uint16_t>(ptr);) =26